hclasses/CDirReader.cpp

```diff
@@ -53,18 +53,30 @@ std::vector<std::string> CDirReader::listDir(const char *directory, bool recursi
   if ((dir = opendir(directory)) != NULL) {
     /* print all the files and directories within directory */
     while ((ent = readdir(dir)) != NULL) {
-      if (ent->d_type == DT_REG || ent->d_type == DT_DIR) {
-        if (ent->d_name[0] != '.') { // Omit dirs starting with a .
-          CT::string fullName = directory;
-          fullName.concat("/");
-          fullName.concat(ent->d_name);
+      CT::string fullName = directory;
+      fullName.concat("/");
+      fullName.concat(ent->d_name);
+      // Deal with filesystems that don't provide d_type
+      auto d_type = ent->d_type;
+      if (d_type == DT_UNKNOWN) {
+        struct stat path_stat{};
+        int ret = stat(fullName.c_str(), &path_stat);
+        if (ret == 0 && S_ISREG(path_stat.st_mode)) {
+          d_type = DT_REG;
+        }
+        if (ret == 0 && S_ISDIR(path_stat.st_mode)) {
+          d_type = DT_DIR;
+        }
+      }
 
-          if (((filesAndOrDirs & CDIRREADER_INCLUDE_FILES) && ent->d_type == DT_REG) || ((filesAndOrDirs & CDIRREADER_INCLUDE_DIRECTORIES) && ent->d_type == DT_DIR)) {
+      if (d_type == DT_REG || d_type == DT_DIR) {
+        if (ent->d_name[0] != '.') { // Omit dirs starting with a .
```

Comment:
Note, the comment says "omit dirs starting with a ." but we are also omitting files starting with a ".". The file `.my-dataset.xml` would pass the regex but still get excluded.

Is this intended behavior?
If yes, then maybe update the comment?